corehq/apps/analytics/static/analytix/js/kissmetrix.js

```diff
@@ -60,11 +61,14 @@ hqDefine('analytics/js/kissmetrics', function () {
             testName = _.last(testName.split('.'));
             if (_.isObject(ab) && ab.version) {
                 test[ab.name || testName] = ab.version;
-            } else {
+            } else if (!_.isEmpty(ab)) {
```

Comment:
It looks like `getAbTests` only returns objects with keys for `slug` and `context`, why does this if/else test if ab is an object and look for ab.name and ab.version, and then potentially treat ab as a string? Could use a comment explaining why this is all needed.

Reply:
i think this is me overengineering 😳